### ISSUE TITLE
Pause playback when switching audio device

### ISSUE THREAD

**Issue by q-wertz** [xou816/spot#235]
**Describe the bug**
Currently when listening with headphones (bluetooth) and switching them off the audio automatically switches to the boxes and the music continues to play.

**To Reproduce**
1. Play music using bluetooth headphones
2. Switch off headphones
3. Music continues to play on boxes

**Expected behavior**
Playback is paused when switching

**General information:**
 - Distribution: Ubuntu 21.04 (Wayland)
 - Installation method: snap (from https://github.com/popey/spot-snap)
 - Version: 0.1.14
 - Device used: desktop

**Additional context**
Current way is a little bit inconvenient as when I'm listening to music during work my collegues maybe do not want to have to listen to the music I do :)

**Comment by xou816** [comment]
HI! :)

Hmm, I have no idea how that would be implemented, especially in a way that does not depend on a specific audio backend :thinking: have you seen this being implemented in any player?

Also, I think this shouldn't be the responsibility of a specific audio player -- instead, maybe some part of the DE would be responsible for monitoring changes in active audio devices and play/pause the active media player over DBUS?

Long story short, this is both beyond my abilities and the scope of the project, sorry, I most likely won't work on that :/ But I wish this existed at the DE level indeed!

**Comment by q-wertz** [comment]
> Hi! :)
> 
> Hmm, I have no idea how that would be implemented, especially in a way that does not depend on a specific audio backend thinking have you seen this being implemented in any player?

I thought so but now I didn't find one where it is that way. Maybe just my "workflow" leaving my Laptop changed :sweat_smile: 

Where it is definitely working is on Android (e.g. Spotify). And there in the best way I can think of: Audio continues when switching from speakers to headphones, audio is stopped when turning headphones off.

> Also, I think this shouldn't be the responsibility of a specific audio player -- instead, maybe some part of the DE would be responsible for monitoring changes in active audio devices and play/pause the active media player over DBUS?
> 
> Long story short, this is both beyond my abilities and the scope of the project, sorry, I most likely won't work on that :/ But I wish this existed at the DE level indeed!

No problem. Thanks for the explanation :blush:
